src/flamegraph.cpp

```diff
@@ -274,10 +300,60 @@ QBrush brushImpl(uint hash, BrushType type)
     return brushes.at(hash % brushes.size());
 }
 
+QBrush brushBinary(const Data::Symbol& symbol)
+{
+    static QHash<QString, QBrush> brushes;
+
+    if (brushes.contains(symbol.binary)) {
+        return brushes[symbol.binary];
+    }
+
+    // make gaps between colors since a human cant see a difference between #aabbff and #aabbfe
+    QBrush _brush(QColor((2 + rand(16)) * 8, (15 + rand(16)) * 8, (15 + rand(16)) * 8, 125));
+    brushes[symbol.binary] = _brush;
+    return _brush;
+}
+
+QBrush brushKernel(const Data::Symbol& symbol)
+{
+    static QBrush kernel(QColor(255, 0, 0, 125));
+    static QBrush user(QColor(0, 0, 255, 125));
+
+    if (symbol.isKernel) {
+        return kernel;
+    }
+    return user;
+}
+
+QBrush brushSystem(const Data::Symbol& symbol)
+{
+    static QBrush system(QColor(0, 125, 0, 125));
+    static QBrush user(QColor(200, 200, 0, 125));
+    static QBrush unkown(QColor(50, 50, 50, 125));
+
+    // I have seen [ only on kernel calls, but am not sure if this is correct
+    if (symbol.path.isEmpty() || symbol.path.startsWith(QLatin1String("["))) {
+        return unkown;
+    } else if (symbol.path.startsWith(QLatin1String("/usr")) || symbol.path.startsWith(QLatin1String("/lib"))) {
```

Comment:
this is bogus and not cross platform compliant (won't work on Windows e.g.). Worse, it won't work at all for custom sysroots and embedded programs. Please add a notion of "system paths" and "user paths" to the `Settings` singleton and UI to configure it (separate patch!). By default the "system paths" should be the `QDir::root` path and "user paths" should be set to the user's home directory. Then the logic should be:

- is the path in any of the system paths, then mark as system
- is the path marked as system but also in a user path, then override and set as user path again